hack/release-snapshot.sh

```diff
@@ -0,0 +1,159 @@
+         exit 1
+}
+
+if [[ -z "${SNAPSHOT_JSON}" ]]; then
+  echo "Error: Finding snapshot for label filter: ${LABEL_FILTER}"
+  exit 1
+fi
+
+COMMIT_SHA_TITLE=$(echo "${SNAPSHOT_JSON}" | jq -r '.metadata.annotations["pac.test.appstudio.openshift.io/sha-title"]')
+echo "Snapshot commit title: ${COMMIT_SHA_TITLE}"
+
+COMMIT_SHA=$(echo "${SNAPSHOT_JSON}" | jq -r '.metadata.annotations["build.appstudio.redhat.com/commit_sha"]')
+echo "Snapshot commit sha: ${COMMIT_SHA}"
+
+COMPONENT_NAME="windows-machine-config-operator-${TARGET_BRANCH//./-}"
+OPERATOR_CONTAINER_COMMIT_SHA=$(echo "${SNAPSHOT_JSON}" | jq -r --arg componentName "$COMPONENT_NAME" '.spec.components[] | select(.name == $componentName) | .source.git.revision')
+if [[ -z "${OPERATOR_CONTAINER_COMMIT_SHA}" ]]; then
+  echo "Error: cannot find source revision for component: ${COMPONENT_NAME}"
+  exit 1
+fi
+if [[ "${COMMIT_SHA}" != "${OPERATOR_CONTAINER_COMMIT_SHA}" ]]; then
+  echo ""
+  echo "Warning: commit sha does not match with operator source revision: ${OPERATOR_CONTAINER_COMMIT_SHA}"
+  OPERATOR_CONTAINER_IMAGE=$(echo "${SNAPSHOT_JSON}" | jq -r --arg componentName "$COMPONENT_NAME" '.spec.components[] | select(.name == $componentName) | .containerImage')
+  echo "Snapshot operator container image: ${OPERATOR_CONTAINER_IMAGE}"
+  echo ""
+fi
+
+COMPONENT_NAME="windows-machine-config-operator-bundle-${TARGET_BRANCH//./-}"
+BUNDLE_CONTAINER_IMAGE=$(echo "${SNAPSHOT_JSON}" | jq -r --arg componentName "$COMPONENT_NAME" '.spec.components[] | select(.name == $componentName) | .containerImage')
+if [[ -z "${BUNDLE_CONTAINER_IMAGE}" ]]; then
+  echo "Error: cannot find containerImage for component: ${COMPONENT_NAME}"
+  exit 1
+fi
+echo "Snapshot bundle container image: ${BUNDLE_CONTAINER_IMAGE}"
+
+# extract the tag from the bundle container image
+BUNDLE_CONTAINER_IMAGE_HASH="${BUNDLE_CONTAINER_IMAGE#*@sha256:}"
+echo "Snapshot bundle container image hash: ${BUNDLE_CONTAINER_IMAGE_HASH}"
+
+# extract the registry and repository from the current image value
+IMAGE_VALUE="${IMAGE_VALUE%@sha256:*}"
+
+# update the image with the new bundle container image
+IMAGE_VALUE="${IMAGE_VALUE}@sha256:${BUNDLE_CONTAINER_IMAGE_HASH}"
+
+# replace the SHA in the catalog template
+jq --indent 4 --arg image_value "$IMAGE_VALUE" '.entries[-1].image = $image_value' \
+  "${CATALOG_TEMPLATE_PATH}/catalog-template.json" > tmp.json && \
+  mv tmp.json "${CATALOG_TEMPLATE_PATH}/catalog-template.json"
+
+# read file and print the updated image value
+jq -r '.entries[-1].image' "${CATALOG_TEMPLATE_PATH}/catalog-template.json"
+echo "Updated olm.bundle image hash in ${CATALOG_TEMPLATE_PATH}/catalog-template.json"
+
+# render template and create the catalog.json file
+opm alpha render-template \
+  basic "${CATALOG_TEMPLATE_PATH}/catalog-template.json" \
+  --migrate-level=bundle-object-to-csv-metadata > "${CATALOG_TEMPLATE_PATH}/catalog/windows-machine-config-operator/catalog.json" || {
```

Comment:
`  --migrate-level=bundle-object-to-csv-metadata`

This should only be used for 4.17 and higher.
See docs: https://github.com/konflux-ci/olm-operator-konflux-sample/blob/main/docs/konflux-onboarding.md#create-the-fbc-in-the-git-repository

Reply:
Addressed in https://github.com/openshift/windows-machine-config-operator-fbc/compare/63b5730bb8d7e729c56bd852b3adac4966cd885e..3513af27d7aa41a087e23918d9ec4171e8c5ad51

Reply:
@sebsoto  PTAL